ng/src/web/components/dialog/dialog.js

```diff
@@ -89,53 +110,105 @@ class Dialog extends React.Component {
     }
   }
 
-  onMouseDown(event) {
+  onMouseDownMove(event) {
     if ((event.buttons & 1) && has_value(this.dialog)) { // eslint-disable-line no-bitwise
       const box = this.dialog.getBoundingClientRect();
       this.relX = event.pageX - box.left;
       this.relY = event.pageY - box.top;
-      document.addEventListener('mousemove', this.onMouseMove);
+
+      document.addEventListener('mousemove', this.onMouseMoveMove);
       document.addEventListener('mouseup', this.onMouseUp);
       event.preventDefault();
     }
   }
 
-  onMouseMove(event) {
+  onMouseDownResize(event) {
+    if (event.buttons & 1) { // eslint-disable-line no-bitwise
+      const box = this.dialog.getBoundingClientRect();
+
+      this.width = Math.round(box.width);
+      this.height = Math.round(box.height);
+      this.mousePosX = event.pageX;
+      this.mousePosY = event.pageY;
+      this.fixedPosX = box.left;
+      this.fixedPosY = box.top;
+
+      document.addEventListener('mousemove', this.onMouseMoveResize);
+      document.addEventListener('mouseup', this.onMouseUp);
+      event.preventDefault();
+    }
+  }
+
+  onMouseMoveMove(event) {
     this.setState({
       posX: event.pageX - this.relX,
       posY: event.pageY - this.relY,
     });
     event.preventDefault();
   }
 
+  onMouseMoveResize(event) {
+    const differenceX = this.mousePosX - event.pageX;
+    const differenceY = this.mousePosY - event.pageY;
+    let newWidth = this.width - differenceX;
+    let newHeight = this.height - differenceY;
+
+    if (newWidth < DEFAULT_DIALOG_MIN_WIDTH) {
+      newWidth = DEFAULT_DIALOG_MIN_WIDTH;
+    }
+    if (newHeight < DEFAULT_DIALOG_MIN_HEIGHT) {
+      newHeight = DEFAULT_DIALOG_MIN_HEIGHT;
+    }
+
+    this.setState({
+      posX: this.fixedPosX,
+      posY: this.fixedPosY,
+      width: newWidth,
+      height: newHeight,
+    });
+    event.preventDefault();
+  }
+
   onMouseUp(event) {
-    document.removeEventListener('mousemove', this.onMouseMove);
+    document.removeEventListener('mousemove', this.onMouseMoveMove);
+    document.removeEventListener('mousemove', this.onMouseMoveResize);
     document.removeEventListener('mouseup', this.onMouseUp);
     event.preventDefault();
   }
 
   getMoveProps() {
     return {
-      onMouseDown: this.onMouseDown,
+      onMouseDown: this.onMouseDownMove,
     };
   }
 
   render() {
-    const {
+    let {
       posX,
       posY,
+      height = DEFAULT_DIALOG_HEIGHT,
+      width = DEFAULT_DIALOG_WIDTH,
     } = this.state;
 
     const {
       children,
+      resizable = true,
       visible,
-      width = DEFAULT_DIALOG_WIDTH,
+      width: setWidth,
     } = this.props;
 
+    if (!resizable) {
+      height = 'auto';
+    }
+
     if (!visible) {
       return null;
     }
 
+    if (is_defined(setWidth)) {
+      width = setWidth;
```

Comment:
IMHO this doesn't work correctly. If the width is set on a dialog resizing the width won't be possible anymore.

To get the expected behavior just set the width in componentDidMount:
```javascript
componentDidMount() {
    document.addEventListener('keydown', this.onKeyDown);

    this.setState({width: this.props.width});
}
```

Afterwards only use the width from the state. Then a user is able to set an initial width when the dialog is rendered the first time. Afterwards changes on the width prop aren't considered anymore. If we need this also in future componentWillReceiveProps has to be implemented. But I don't think so at the moment.